### PR TITLE
feat: Added lightsail databases

### LINKED ISSUE #773: AWS::Lightsail::Database

### PR DESCRIPTION
**Pull request by amanenk**
#### Summary

closes #773 

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [x] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [x] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: resources/services/lightsail/databases.go [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+				Description: "The Availability Zone",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Location.AvailabilityZone"),
+			},
+			{
+				Name:        "region_name",
+				Description: "The AWS Region name",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Location.RegionName"),
+			},
```
Comment (bbernays) [inline resources/services/lightsail/databases.go:102]: Seems to be a duplicate of `region`

File: resources/services/lightsail/databases.go [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+	svc := c.Services().Lightsail
+	streams, err := svc.GetRelationalDatabaseLogStreams(ctx, &input, func(options *lightsail.Options) {
+		options.Region = c.Region
+	})
+	if err != nil {
+		return diag.WrapError(err)
+	}
+	endTime := time.Now()
+	startTime := endTime.Add(-time.Hour * 24 * 14) //two weeks
+	var sem = semaphore.NewWeighted(int64(MAX_GOROUTINES))
```
Comment (irmatov) [inline resources/services/lightsail/databases.go:459]: nit: new errgroup has SetLimit method which frees us from using semaphore and simplifies the code.

File: resources/services/lightsail/databases.go [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+	errs, ctx := errgroup.WithContext(ctx)
+	for _, s := range streams.LogStreams {
+		if err := sem.Acquire(ctx, 1); err != nil {
+			return diag.WrapError(err)
```
Comment (irmatov) [inline resources/services/lightsail/databases.go:463]: just `break` here, we'll get a proper error from `errs.Wait()`

Comment (irmatov) [reply resources/services/lightsail/databases.go:463]: ^ this will be irrelevant if we use errgroup's `SetLimit`

**Review by irmatov — CHANGES_REQUESTED** [review]

**Review by irmatov — APPROVED** [review]

**Pull request merged**